Add files via upload

nikitaSua opened the pull request.

nikitaSua committed:
Add files via upload

DaniilManita commented on LInqHomeWork/LinqRequests.cs:
```diff
@@ -0,0 +1,236 @@
+        {
+            Console.WriteLine(System.Reflection.MethodInfo.GetCurrentMethod().Name);
+            
+            var films = list.Where(a=>a is Film).Cast<Film>();
+            var actors = films.Select(x => string.Join(", ", x.Actors.Select(a => a.Name)));
```
говорил же юзать офтайп, и это можно написать в один запрос

DaniilManita commented on LInqHomeWork/LinqRequests.cs:
```diff
@@ -0,0 +1,236 @@
+        {
+            Console.WriteLine($"\n{System.Reflection.MethodInfo.GetCurrentMethod().Name}");
+
+            var films = list.Where(a => a is Film).Cast<Film>();
+            IEnumerable<Actor> actors = films.FirstOrDefault().Actors.Concat(films.LastOrDefault().Actors).Distinct();
+            int restlt = actors.Where(x => x.Birthdate.Month == 8).Count();
```
это все должно быть написанно в одном запросе

nikitaSua committed:
ДЗ по Linq , фикс замечаний 

Объединил все запросы в 1 и поменял конструкцию "List<obj>.where(x=>x is className).cast<className>()"  на  "List<obj>,OfType<ClassName>();"
